app/controllers/user_avatars_controller.rb

```diff
@@ -203,14 +203,16 @@ def get_optimized_image(upload, size)
   def fix_extension(upload)
     # this is relatively cheap
     original_path = Discourse.store.path_for(upload)
+
     if original_path.blank?
-      external_copy = Discourse.store.download(upload) rescue nil
```

Comment:
I've been bitten by `rescue nil` many times. Any typos or undefined method are buried so I don't think we should use the pattern.

Reply:
happy to have this fixed AS LONG as its also fixed in UploadCreator

Reply:
oh was commenting on fast image ... this particular code is from OptimizedImage, so it needs to also be fixed